src/ansi-c/padding.cpp

```diff
@@ -201,6 +206,8 @@ static void add_padding_msvc(struct_typet &type, const namespacet &ns)
         offset += (bit_field_bits + pad_bits) / config.ansi_c.char_width;
         underlying_bits = bit_field_bits = 0;
       }
+      else
+        offset += underlying_bits / config.ansi_c.char_width;
```

Comment:
I think this is wrong: for example, `struct { char x : 4, y : 4, z : 4, w : 4 }` would have a train of 4 bitfields with underlying width 8, would get to the end of the bitfield sequence and then add 1 (underlying bits == 8 / char_width) rather than 2 to offset as it ought to. This probably ought to share the offset code with the case above, for the special case where `pad_bits == 0`?

Reply:
Thanks a lot, fixed and regression test based on your example added as well.